contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -574,6 +579,41 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
         _setContractURI(_newContractURI);
     }
 
+    /** @notice Make a call to an external contract.
+     *
+     * Reverts if:
+     * - _to is zero address
+     * - call to external contract fails
+     * - caller is not the owner
+     *
+     * @param _to - address of the contract to call
+     * @param _data - data to pass to the external contract
+     */
+    function callExternalContract(address _to, bytes memory _data) external payable onlyOwner {
+        require(_to != address(0), INVALID_ADDRESS);
+
+        _to.functionCallWithValue(_data, msg.value, FUNCTION_CALL_NOT_SUCCESSFUL);
+    }
+
+    /** @notice Set approval for all to the vouchers owned by this contract
+     *
+     * Reverts if:
+     * - _operator is zero address
+     * - caller is not the owner
+     * - _operator is this contract
+     *
+     * @param _operator - address of the operator to set approval for
+     * @param _approved - true if the operator is approved, false to revoke approval
+     */
+    function setApprovalForAllToContract(address _operator, bool _approved) external onlyOwner {
+        require(_operator != address(0), INVALID_ADDRESS);
+
+        _setApprovalForAll(address(this), _operator, _approved);
+    }
+
+    // @dev Contract must be allowed to receive native token as it can be used as voucher's owner
+    receive() external payable {}
```

Comment:
Given that, in this PR, we have made the contract payable, do we want to make it an ERC721 receiver too ?

See this file ?

https://github.com/OpenZeppelin/openzeppelin-contracts/tree/4fb6833e325658946c2185862b8e57e32f3683bc/contracts/token/ERC721/IERC721Receiver.sol

Reply:
added